projects/gateway2/translator/sslutils/ssl_utils.go

```diff
@@ -45,5 +47,20 @@ func isValidSslKeyPair(certChain, privateKey, rootCa []byte) error {
 	}
 
 	_, err := tls.X509KeyPair([]byte(certChain), []byte(privateKey))
+	if err != nil {
+		return err
+	}
+	candidateCert, err := cert.ParseCertsPEM([]byte(certChain))
+	if err != nil {
+		// return err rather than sanitize. This is to maintain UX with older versions and to prevent a large refactor
+		// for gatewayv2 secret validation code.
```

Comment:
If I understand correctly from the linked gist, by "sanitize" we mean "attempt to extract valid blocks even if the certChain is not entirely valid"

I think we can improve the comment here
```suggestion
		// return err rather than attempt to extract and return valid certs from the certChain.
		// while doing so may be desirable in some circumstances, returning when any cert is invalid maintains consistent
		// UX with older versions and to prevent a large refactor for gatewayv2 secret validation code.
```

Reply:
this isnt locked to cets and we do have a comment that states what does or doesnt work.

This comment is purely around why we dont update our certchain to being good rather than just error out.